presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergSplitManager.java

```diff
@@ -58,8 +68,14 @@ public ConnectorSplitSource getSplits(
             return new FixedSplitSource(ImmutableList.of());
         }
 
-        ExtendedHiveMetastore metastore = transactionManager.get(transaction).getMetastore();
-        Table icebergTable = getIcebergTable(metastore, hdfsEnvironment, session, table.getSchemaTableName());
+        Table icebergTable;
+        if (nativeCatalogMode) {
+            icebergTable = resourceFactory.getCatalog(session).loadTable(toIcebergTableIdentifier(table.getSchemaTableName()));
```

Comment:
although this is reloading the Iceberg table, because we always input the actual snapshot ID, snapshot isolation is still guaranteed. Some optimizations can be made to cache the table object, but I am not adding that here to keep things simple.